pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/LmsInsertDsmEvents.java

```diff
@@ -0,0 +1,58 @@
+package org.broadinstitute.ddp.studybuilder.task;
+
+import lombok.extern.slf4j.Slf4j;
+import one.util.streamex.StreamEx;
+import org.broadinstitute.ddp.db.dao.EventDao;
+import org.broadinstitute.ddp.db.dao.JdbiEventConfiguration;
+import org.broadinstitute.ddp.db.dao.JdbiUmbrellaStudy;
+import org.broadinstitute.ddp.model.activity.types.EventActionType;
+import org.broadinstitute.ddp.model.activity.types.EventTriggerType;
+import org.broadinstitute.ddp.model.dsm.DsmNotificationEventType;
+import org.broadinstitute.ddp.model.event.DsmNotificationTrigger;
+import org.broadinstitute.ddp.model.event.EventConfiguration;
+import org.jdbi.v3.core.Handle;
+
+@Slf4j
+public class LmsInsertDsmEvents extends InsertStudyEvents {
+    public LmsInsertDsmEvents() {
+        super("cmi-lms", "patches/lms-dsm-events.conf");
+    }
+
+    @Override
+    public void run(final Handle handle) {
+        removeExistingEvents(handle);
+        super.run(handle);
+    }
+
+    private void removeExistingEvents(final Handle handle) {
+        final var eventDao = handle.attach(EventDao.class);
+        final var count = StreamEx.of(eventDao.getAllEventConfigurationsByStudyId(handle.attach(JdbiUmbrellaStudy.class)
+                        .findByStudyGuid(studyGuid).getId()))
+                .filterBy(EventConfiguration::getEventActionType, EventActionType.ACTIVITY_INSTANCE_CREATION)
+                .filterBy(EventConfiguration::getEventTriggerType, EventTriggerType.DSM_NOTIFICATION)
+                .filter(this::hasDSMNotificationTrigger)
+                .filter(this::isExpectedTrigger)
+                .map(EventConfiguration::getEventConfigurationId)
+                .map(id -> deactivateEventConfiguration(handle, id))
+                .count();
+
+        log.info("Successfully deactivated {} DSM Notification events that create new activities of {}.", count, studyGuid);
```

Comment:
Should there be explicit checks on the count so we make sure that we're removing the right number of things instead of just writing something out to the log?

Reply:
Since the query is not checking already disabled/inactive events, usually count is different in Dev.
Moreover , we are just disabling and not actually removing. I should have called the method as disableExistingEvents.  I usually check those displayed events ids by querying in DB.